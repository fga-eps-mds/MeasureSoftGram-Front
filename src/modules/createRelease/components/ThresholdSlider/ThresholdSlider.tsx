import React from 'react';
import { Box, Grid, Slider, TextField, Typography } from '@mui/material';
import getThresholdInfo from '@modules/createRelease/utils/getThresholdInfo';

interface PreConfigSlidersProps {
  label: string;
  onChange: (min: number, max: number, key: string) => void;
  min: number;
  max: number;
}

const ThresholdSlider = ({ label, onChange, min, max }: PreConfigSlidersProps) => {
  function changeSlider(e: Event) {
    const value = e.target!.value as unknown as number[];
    onChange(value[0], value[1], label);
  }

  const thresholdInfo = getThresholdInfo(label);

  return (
    <Box display="flex" flexDirection="column">
      <Typography textAlign="center">
        <strong>{thresholdInfo?.label ?? ''}</strong>
      </Typography>
      <Typography textAlign="center">{thresholdInfo?.description ?? ''}</Typography>
      <Grid spacing={3} container columns={16}>
        <Grid item xs={2}>
          <TextField
            disabled={thresholdInfo?.minFixed ?? false}
            sx={{ top: -4 }}
            type="number"
            variant="standard"
            value={min}
            onChange={(e) => onChange(Number(e.target.value), max, label)}
          />
        </Grid>
        <Grid item xs={12}>
          {(thresholdInfo?.range[1] ?? undefined) !== undefined && (
            <Slider
              value={[min, max]}
              onChange={(e) => changeSlider(e)}
              min={thresholdInfo?.range[0] ?? 0}
              max={thresholdInfo?.range[1] ?? 100}
              step={thresholdInfo?.step}
              data-testid="threshold-slider"
            />
          )}
        </Grid>
        <Grid item xs={2}>
          <TextField
            disabled={thresholdInfo?.maxFixed ?? false}
            sx={{ top: -4 }}
            variant="standard"
            type="number"
            value={max}
            onChange={(e) => onChange(min, Number(e.target.value), label)}
            data-testid="threshold-slider-max"
          />
        </Grid>
      </Grid>
    </Box >
  );
};

export default ThresholdSlider;
