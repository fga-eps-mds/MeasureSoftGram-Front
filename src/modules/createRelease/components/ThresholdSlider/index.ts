import ThresholdSlider from './ThresholdSlider';

export default ThresholdSlider;
