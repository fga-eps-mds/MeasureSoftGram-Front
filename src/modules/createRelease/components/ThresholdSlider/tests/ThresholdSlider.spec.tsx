import { fireEvent, render } from "@testing-library/react";
import React from "react";
import ThresholdSlider from "../ThresholdSlider";

interface SliderMockProps {
  onChange: (event: any) => void;
}

const SliderMock = ({ onChange }: SliderMockProps) => (
  <input data-testid="threshold-slider" onChange={onChange} />);

interface TextFieldMockProps {
  onChange: (event: any) => void;
}
const TextFieldMock = ({ onChange }: TextFieldMockProps) => <input data-testid="threshold-textfield" onChange={onChange} />;

jest.mock("@mui/material/Slider", () => SliderMock);

jest.mock("@mui/material/TextField", () => TextFieldMock);

describe("<ThresholdSlider />", () => {
  describe("Snapshot", () => {
    it("Deve corresponder ao snapshot", () => {
      const tree = render(<ThresholdSlider label="duplication_absense" onChange={jest.fn()} min={0} max={100} />);
      expect(tree).toMatchSnapshot();
    })
  });

  it("Deve chamar changeSlider ao alterar o valor do slider", () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<ThresholdSlider label="duplication_absense" onChange={onChange} min={0} max={100} />);
    const sliderInput = getByTestId("threshold-slider");
    expect(sliderInput).toBeDefined();
    fireEvent.change(sliderInput, { target: { value: [50, 80] } })
    expect(onChange).toBeCalled();
  })

  it("Deve chamar onChange ao alterar o valor do textfield", () => {
    const onChange = jest.fn();
    const { getAllByTestId } = render(<ThresholdSlider label="duplication_absense" onChange={onChange} min={0} max={100} />);
    const textFieldInputs = getAllByTestId("threshold-textfield");
    expect(textFieldInputs[0]).toBeDefined();
    expect(textFieldInputs[1]).toBeDefined();
    fireEvent.change(textFieldInputs[0], { target: { value: 50 } })
    fireEvent.change(textFieldInputs[1], { target: { value: 80 } })
    expect(onChange).toBeCalled();
  })
});
