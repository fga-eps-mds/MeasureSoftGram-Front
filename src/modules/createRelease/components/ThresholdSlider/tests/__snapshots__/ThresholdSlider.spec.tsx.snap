// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThresholdSlider /> Snapshot Deve corresponder ao snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
        >
          <strong>
            Linhas duplicadas
          </strong>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
        >
          min fixo em 0, max entre 0-100
        </p>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1duidt6-MuiGrid-root"
          >
            <input
              data-testid="threshold-textfield"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1qpzhgb-MuiGrid-root"
          >
            <input
              data-testid="threshold-slider"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1duidt6-MuiGrid-root"
          >
            <input
              data-testid="threshold-textfield"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-j7qwjs"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
      >
        <strong>
          Linhas duplicadas
        </strong>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
      >
        min fixo em 0, max entre 0-100
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1duidt6-MuiGrid-root"
        >
          <input
            data-testid="threshold-textfield"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1qpzhgb-MuiGrid-root"
        >
          <input
            data-testid="threshold-slider"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1duidt6-MuiGrid-root"
        >
          <input
            data-testid="threshold-textfield"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
